### PR TITLE
feat: ✨ adds confirmation modals, alerts, and notes to Approve Agreement page

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- alerts and confirmation modals for Cancel, Decline, and Approve buttons
- adds notes to alerts
- adds submitter's notes

## Issue

#2349 

## How to test

1. Goto Agreement 1 or 9
2. make a status or budget change
3. send to approval
4. Goto Review list and click on review card to see approval page
5. Approve/Decline/Cancel buttons actions will trigger a confirmation modal and Alert passing along the notes and change requests

## Screenshots

https://github.com/user-attachments/assets/f851337e-e43c-456b-a981-b8d8e1ff0fa1

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

**Comment by fpigeonjr** [comment]
🥊 opening up for review. Will update e2e tests for this round before merging in.

File: frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js [hunk 1]
```diff
@@ -97,23 +113,62 @@ const useApproveAgreement = () => {
...
-        statusForTitle = `- ${renderField(null, "status", status)}`;
+        statusForTitle = `- ${renderField(null, "status", statusChangeTo)}`;
     }
     const changeRequestTitle = toTitleCaseFromSlug(changeRequestType);
     const title = `Approval for ${changeRequestTitle} ${statusForTitle}`;
 
+    let requestorNoters = "";
+    if (changeRequestType !== CHANGE_REQUEST_SLUG_TYPES.BUDGET) {
+        requestorNoters = changeRequestsInReview[0]?.requestor_notes ?? "";
+    }
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js:133]: @stevtek is it a safe assumption to just grab the submitters_notes from the first in_review BLI that is not of budget change?

Comment (stevtek) [reply frontend/src/pages/agreements/approve/ApproveAgreement.hooks.js:133]: I think that collecting all distinct notes is better

**Comment by fpigeonjr** [comment]
- 🟢 e2e test have been updates/added

File: frontend/cypress/e2e/approveChangeRequestsAtAgreementLevel.cy.js [hunk 2]
```diff
@@ -433,6 +488,181 @@
...
+                            method: "DELETE",
+                            url: `http://localhost:8080/api/v1/agreements/${agreementId}`,
+                            headers: {
+                                Authorization: bearer_token,
+                                Accept: "application/json"
+                            },
+                            failOnStatusCode: false
+                        }).then((response) => {
+                            if (response.status !== 200) {
+                                cy.log(`Failed to delete agreement ${agreementId}. Status: ${response.status}`);
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/cypress/e2e/approveChangeRequestsAtAgreementLevel.cy.js]: reverted test back

**Review by johndeange — APPROVED** [review]

**Pull request merged**